### PR TITLE
UAR-1610 Introduce new nocs to GPA BO and validate

### PR DESCRIPTION
**Pull request by mattch1**
### JIRA link
https://companieshouse.atlassian.net/browse/UAR-1610

### Change description

Similar to and using the changes in:
https://github.com/companieshouse/ecs-service-configs-dev/pull/621
updated the model and validator with the three new fields.

### Work checklist

- [x] Tests added where applicable

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dao/BeneficialOwnerGovernmentOrPublicAuthorityDao.java [hunk 1]
```diff
@@ -35,6 +36,18 @@ public class BeneficialOwnerGovernmentOrPublicAuthorityDao {
     @Field("non_legal_firm_members_nature_of_control_types")
     private List<NatureOfControlType> nonLegalFirmMembersNatureOfControlTypes;
 
+    @Field("non_legal_firm_control_nature_of_control_types")
+    private List<NatureOfControlType> nonLegalFirmControlNatureOfControlTypes;
+
+    @Field("trust_control_nature_of_control_types")
+    private List<NatureOfControlType> trustNatureOfControlTypes;
```
Comment (markpit) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dao/BeneficialOwnerGovernmentOrPublicAuthorityDao.java:43]: trustNatureOfControlTypes is missing 'Control' in the attribute name (line 43) - trustControlNatureOfControlTypes

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dao/BeneficialOwnerGovernmentOrPublicAuthorityDao.java [hunk 2]
```diff
@@ -125,6 +138,38 @@ public void setNonLegalFirmMembersNatureOfControlTypes(List<NatureOfControlType>
...
 
+    public List<NatureOfControlType> getNonLegalFirmControlNatureOfControlTypes() {
+        return nonLegalFirmControlNatureOfControlTypes;
+    }
+
+    public void setNonLegalFirmControlNatureOfControlTypes(List<NatureOfControlType> nonLegalFirmControlNatureOfControlTypes) {
+        this.nonLegalFirmControlNatureOfControlTypes = nonLegalFirmControlNatureOfControlTypes;
+    }
+
+    public List<NatureOfControlType> getTrustNatureOfControlTypes() {
```
Comment (markpit) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dao/BeneficialOwnerGovernmentOrPublicAuthorityDao.java:149]: the getter and setter will need renaming to include the 'Control' - getTrustControlNatureOfControlTypes and set..

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dto/BeneficialOwnerGovernmentOrPublicAuthorityDto.java [hunk 3]
```diff
@@ -17,6 +18,10 @@ public class BeneficialOwnerGovernmentOrPublicAuthorityDto {
     public static final String START_DATE_FIELD = "start_date";
     public static final String BENEFICIAL_OWNER_NATURE_OF_CONTROL_TYPES_FIELD = "beneficial_owner_nature_of_control_types";
     public static final String NON_LEGAL_FIRM_MEMBERS_NATURE_OF_CONTROL_TYPES_FIELD = "non_legal_firm_members_nature_of_control_types";
+    public static final String NON_LEGAL_FIRM_CONTROL_NATURE_OF_CONTROL_TYPES_FIELD = "non_legal_firm_control_nature_of_control_types";
+    public static final String TRUST_NATURE_OF_CONTROL_TYPES_FIELD = "trust_control_nature_of_control_types";
```
Comment (markpit) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dto/BeneficialOwnerGovernmentOrPublicAuthorityDto.java:22]: the static string needs updating to TRUST_CONTROL_NATURE_OF_CONTROL_TYPES_FIELD

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dto/BeneficialOwnerGovernmentOrPublicAuthorityDto.java [hunk 4]
```diff
@@ -50,6 +55,18 @@ public class BeneficialOwnerGovernmentOrPublicAuthorityDto {
     @JsonProperty(NON_LEGAL_FIRM_MEMBERS_NATURE_OF_CONTROL_TYPES_FIELD)
     private List<NatureOfControlType> nonLegalFirmMembersNatureOfControlTypes;
 
+    @JsonProperty(NON_LEGAL_FIRM_CONTROL_NATURE_OF_CONTROL_TYPES_FIELD)
+    private List<NatureOfControlType> nonLegalFirmControlNatureOfControlTypes;
+
+    @JsonProperty(TRUST_NATURE_OF_CONTROL_TYPES_FIELD)
+    private List<NatureOfControlType> trustNatureOfControlTypes;
```
Comment (markpit) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dto/BeneficialOwnerGovernmentOrPublicAuthorityDto.java:62]: trustNatureOfControlTypes should be trustControlNatureOfControlTypes

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dto/BeneficialOwnerGovernmentOrPublicAuthorityDto.java [hunk 5]
```diff
@@ -137,6 +154,38 @@ public void setNonLegalFirmMembersNatureOfControlTypes(List<NatureOfControlType>
...
 
+    public List<NatureOfControlType> getNonLegalFirmControlNatureOfControlTypes() {
+        return nonLegalFirmControlNatureOfControlTypes;
+    }
+
+    public void setNonLegalFirmControlNatureOfControlTypes(List<NatureOfControlType> nonLegalFirmControlNatureOfControlTypes) {
+        this.nonLegalFirmControlNatureOfControlTypes = nonLegalFirmControlNatureOfControlTypes;
+    }
+
+    public List<NatureOfControlType> getTrustNatureOfControlTypes() {
```
Comment (markpit) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dto/BeneficialOwnerGovernmentOrPublicAuthorityDto.java:165]: getter and setter need TrustControl added

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dao/BeneficialOwnerGovernmentOrPublicAuthorityDao.java (hunk 1 above)
Comment (mattch1) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dao/BeneficialOwnerGovernmentOrPublicAuthorityDao.java:43]: Does this mean we need to add "control" to the existing:
TRUSTEES_NATURE_OF_CONTROL_TYPES_FIELD
as well? This is what I took my lead from in naming this field.
I only added the extra control to firm to distinguish it clearly from the member field.

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dao/BeneficialOwnerGovernmentOrPublicAuthorityDao.java (hunk 2 above)
Comment (mattch1) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dao/BeneficialOwnerGovernmentOrPublicAuthorityDao.java:149]: See question above.

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dto/BeneficialOwnerGovernmentOrPublicAuthorityDto.java (hunk 3 above)
Comment (mattch1) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dto/BeneficialOwnerGovernmentOrPublicAuthorityDto.java:22]: As mentioned above would this have to apply to:
TRUSTEES_CONTROL_NATURE_OF_CONTROL_TYPES_FIELD ?

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dto/BeneficialOwnerGovernmentOrPublicAuthorityDto.java (hunk 5 above)
Comment (mattch1) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dto/BeneficialOwnerGovernmentOrPublicAuthorityDto.java:165]: See question above.

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dto/BeneficialOwnerGovernmentOrPublicAuthorityDto.java (hunk 4 above)
Comment (mattch1) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dto/BeneficialOwnerGovernmentOrPublicAuthorityDto.java:62]: See question above.

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dao/BeneficialOwnerGovernmentOrPublicAuthorityDao.java (hunk 1 above)
Comment (markpit) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dao/BeneficialOwnerGovernmentOrPublicAuthorityDao.java:43]: perhaps, i'm not sure.
But the field coming in via json will be
"trust_control_nature_of_control_types": {
            "type": "array",
            "items": {
              "$ref": "#/components/schemas/NatureOfControlType"
            }

if we leave it as trustNatureOfControlTypes, then that reads as if it is all the Nocs for the trust

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dto/BeneficialOwnerGovernmentOrPublicAuthorityDto.java (hunk 3 above)
Comment (markpit) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dto/BeneficialOwnerGovernmentOrPublicAuthorityDto.java:22]: ideally it should be as close as possible to the json field name

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dao/BeneficialOwnerGovernmentOrPublicAuthorityDao.java (hunk 1 above)
Comment (mattch1) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dao/BeneficialOwnerGovernmentOrPublicAuthorityDao.java]: In light of the json name this has been updated.

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dao/BeneficialOwnerGovernmentOrPublicAuthorityDao.java (hunk 2 above)
Comment (mattch1) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dao/BeneficialOwnerGovernmentOrPublicAuthorityDao.java]: Updated

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dto/BeneficialOwnerGovernmentOrPublicAuthorityDto.java (hunk 3 above)
Comment (mattch1) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dto/BeneficialOwnerGovernmentOrPublicAuthorityDto.java]: updated

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dto/BeneficialOwnerGovernmentOrPublicAuthorityDto.java (hunk 4 above)
Comment (mattch1) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dto/BeneficialOwnerGovernmentOrPublicAuthorityDto.java]: updated

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dto/BeneficialOwnerGovernmentOrPublicAuthorityDto.java (hunk 5 above)
Comment (mattch1) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dto/BeneficialOwnerGovernmentOrPublicAuthorityDto.java]: updated

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by markpit — APPROVED** [review]

**Pull request merged**